src/test/java/org/broadinstitute/hellbender/tools/IntegrationTestSpec.java

```diff
@@ -161,20 +163,35 @@ public static void assertMatchingFiles(List<File> resultFiles, List<String> expe
             String expectedFileName = expectedFiles.get(i);
             File expectedFile = new File(expectedFileName);
             if (expectedFileName.endsWith(".bam")){
-                compareBamFiles(resultFile, expectedFile);
+                assertEqualBamFiles(resultFile, expectedFile);
             } else {
-                compareTextFiles(resultFile, expectedFile);
+                assertEqualTextFiles(resultFile, expectedFile);
             }
         }
     }
 
-    public static void compareTextFiles(File resultFile, File expectedFile) throws IOException {
-        List<String> actualLines = new XReadLines(resultFile).readLines();
-        List<String> expectedLines = new XReadLines(expectedFile).readLines();
+    public static void assertEqualTextFiles(final File resultFile, final File expectedFile) throws IOException {
+        assertEqualTextFiles(resultFile, expectedFile, null);
+    }
+
+    /**
+     * Compares two text files and ignores all lines that start with the comment prefix.
+     */
+    public static void assertEqualTextFiles(final File resultFile, final File expectedFile, final String commentPrefix) throws IOException {
+        final Predicate<? super String> startsWithComment;
+        if (commentPrefix == null){
+            startsWithComment = s -> false;
+        } else {
+            startsWithComment = s -> s.startsWith(commentPrefix);
+        }
+        final List<String> actualLines = new XReadLines(resultFile).readLines().stream().filter(startsWithComment.negate()).collect(Collectors.toList());
+        final List<String> expectedLines = new XReadLines(expectedFile).readLines().stream().filter(startsWithComment.negate()).collect(Collectors.toList());
+
```

Comment:
If you think the semantics are too surprising then it's not a good idea.  I'm worried about the case where someone produces two files, expects them to have the same thing in them, but there's some bug where they are both empty instead.  Probably a pretty rare case though since usually one file is a pre-generated static file.